File: 6kyu/Elixir/StopgninnipSMysdroW.ex
defmodule StopgninnipSMysdroW do
  def fun(str) do
    String.split(str, " ", trim: true)
    |> Enum.map(fn x -> 
      if(String.length(x) >= 5) do
        String.reverse(x)
      else
        x
      end
    end)
    |> Enum.join(" ")
  end
end

File: CodeWars/6kyu/Java/StopgninnipSMysdroW.java
public class StopgninnipSMysdroW{
  public static void main(String [] args){
    System.out.println(fun("Patata") + "\n");
    System.out.println(fun("tapa") + "\n");
    System.out.println(fun("Hey fellow warriors") + "\n");
    System.out.println(fun("This is a test") + "\n");
    System.out.println(fun("This is another test") + "\n");
  }

  public static String fun(String str){
    String [] wList = str.split(" ");
    String res = "";

    for(String word: wList){
      if(word.length() >= 5){
        res += reverse(word);
      }
      else{
        res += word;
      }
      res += " ";
    }

    return res.strip();
  }

  public static String reverse(String word){
    String reversed = "";
    for(int i = (word.length() - 1); i >= 0; i--){
      reversed += word.charAt(i);
    }

    return reversed;
  }
}
